pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java

```diff
@@ -204,11 +204,7 @@ protected MseBlock getNextBlock() {
     try {
       MseBlock block = _input.nextBlock();
       if (block.isEos()) {
-        if (_context.isSendStats()) {
-          sendEos((MseBlock.Eos) block);
-        } else {
-          sendEos(SuccessMseBlock.INSTANCE);
-        }
```

Comment:
This was erroneous. In case the block was an error and isSendStats was false, we sent a successful EOS.